src/core/lib/gpr/env_linux.cc

```diff
@@ -40,22 +40,7 @@
 
 char* gpr_getenv(const char* name) {
   char* result = nullptr;
-#if defined(GPR_BACKWARDS_COMPATIBILITY_MODE)
-  typedef char* (*getenv_type)(const char*);
-  static getenv_type getenv_func = nullptr;
-  /* Check to see which getenv variant is supported (go from most
-   * to least secure) */
-  if (getenv_func == nullptr) {
```

Comment:
I think @nicolasnoble  has initially came up with this workaround so he should be able to tell if this is ok to remove (I don't have that much context, but I'm guessing it's fine).